src/main/java/com/epam/university/java/core/task003/FlatMappingOperationImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.epam.university.java.core.task003;
+
+public class FlatMappingOperationImpl implements FlatMappingOperation {
+    @Override
+    public String[] flatMap(String source) {
+        int sizeOfOutput = 0;
+        String outputStr = "";
+        for (int i = 0; i < source.length(); i++) {
+            if (source.charAt(i) >= '0' && source.charAt(i) <= '9') {
+                sizeOfOutput++;
+                do {
+                    outputStr += source.charAt(i);
+                    i++;
+                } while (i < source.length() && source.charAt(i) >= '0' && source.charAt(i) <= '9');
+                outputStr += " ";
+            }
+        }
+        String[] preOutput = new String[sizeOfOutput];
+        int j = 0;
+        for (int i = 0; i < sizeOfOutput; i++) {
+            preOutput[i] = "";
+            while (j < outputStr.length() && outputStr.charAt(j) != ' ') {
+                preOutput[i] += outputStr.charAt(j++);
+            }
+            j++;
+        }
+        String[] output = new String[0];
+        String[] temp;
+        Task003Impl assistant = new Task003Impl();
+        for (int i = 0; i < preOutput.length; i++) {
+            temp = new String[] {""};
+            temp[0] = preOutput[i];
+            temp = assistant.removeElements(temp, output);
```

Comment:
If you need utility methods, it's better to have separated class with static methods for it.